src/Microsoft.Health.Dicom.Core/Features/Model/VersionedInstanceIdentifier.cs

```diff
@@ -12,8 +14,19 @@ public class VersionedInstanceIdentifier : InstanceIdentifier
         string seriesInstanceUid,
         string sopInstanceUid,
         long version,
-        int partitionKey = default)
-        : base(studyInstanceUid, seriesInstanceUid, sopInstanceUid, partitionKey)
+        PartitionEntry partitionEntry)
+        : base(studyInstanceUid, seriesInstanceUid, sopInstanceUid, partitionEntry)
+    {
+        Version = version;
+    }
+    public VersionedInstanceIdentifier(
+        string studyInstanceUid,
+        string seriesInstanceUid,
+        string sopInstanceUid,
+        long version,
+        int partitionKey = default,
+        string partitionName = default)
```

Comment:
Why do you need partitionName? At the time of implementation, we wanted to keep the key instead of name everywhere.